### PR TITLE
[Publisher] Improve toggle experience for metrics page

### LINKED ISSUE #1032: Improve toggle experience for metrics page
For agencies with multiple sectors (such as [this one](https://publisher-staging.justice-counts.org/agency/153/data?system=supervision&metric=supervision_total_staff&view=Count)) it's a pain to switch metrics in the Explore Data page. Users need to scroll down to change the metrics and then scroll back up to see the chart. 

As a solution, we should allow the user to scroll through the list, while the graph is **fixed** on the page.

You can see what it should look like in this [Figma](https://www.figma.com/file/dvX3yg7FMGJSWFaqHWrWgQ/Justice-Counts-Publisher?type=design&node-id=9956%3A982&mode=design&t=sXza8PnsBrXAmeVJ-1). The sidebar should be scrollable on its own, and there is a fading white treatment at the bottom to indicate that.

### PR DESCRIPTION
**Pull request by nasaownsky**
## Description of the change

1. added `react-intersection-observer` package
2. made right container sticky
3. added top and bottom scroll shadows to left container

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

closes #1032 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [ ] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by nasaownsky** [comment]
@lilidworkin @mxosman Hi, team! Because of the way the app works (specifically adding overflow style property to body in many various places), I think the only method we could create expected behavior is to make right side of metrics page (graph and top buttons) sticky. 

Also I added top scroll shadow, I think it adds consistency to the left side of page. Anyway please check my changes and let me know your thoughts on it, thank you!

**Comment by mxosman** [comment]
> @lilidworkin @mxosman Hi, team! Because of the way the app works (specifically adding overflow style property to body in many various places), I think the only method we could create expected behavior is to make the right side of the metrics page (graph and top buttons) sticky.
> 
> Also I added top scroll shadow, I think it adds consistency to the left side of the page. Anyway please check my changes and let me know your thoughts on it, thank you!

Hi @nasaownsky! Thank you so much for tackling this (and for working around the constraints caused by all of our previous decisions). I'm looking forward to reviewing and playing with this later today or tomorrow at the latest!

File: publisher/src/components/DataViz/MetricsDataChart.tsx [hunk 1]
```diff
@@ -39,6 +42,7 @@ import { frequencyString } from "@justice-counts/common/utils/helperUtils";
 import FileSaver from "file-saver";
 import { observer } from "mobx-react-lite";
 import React, { useCallback, useEffect, useState } from "react";
+import { useInView } from "react-intersection-observer";
```
Comment (mxosman) [inline publisher/src/components/DataViz/MetricsDataChart.tsx:45]: Awesome package!

File: publisher/src/components/DataViz/MetricsDataChart.tsx [hunk 2]
```diff
@@ -245,6 +251,12 @@ export const MetricsDataChart: React.FC = observer(() => {
         {/* List Of Metrics */}
         <Styled.PanelContainerLeft>
           <Styled.SystemsContainer>
+            <div ref={topShadow.ref} />
```
Comment (mxosman) [inline publisher/src/components/DataViz/MetricsDataChart.tsx:254]: Are these `div`s the references the intersection observer is watching for?

**Review by mxosman — COMMENTED** [review]
Amazing, Ilya! I love your decisions here, especially working around our current set up. Love the top shadow inclusion as well - thank you so much for that! Looks and feels super smooth!

The only thing standing out to me right now (and this is likely from how it was originally set up) is when scrolling to the bottom, the chart starts to dip away into the button row div (see video). Let me know if you have any thoughts/solutions on making the chart more fixed while scrolling down to the bottom.

https://github.com/Recidiviz/justice-counts/assets/59492998/051e9395-2a3c-405a-ab2b-f4840bdd7e98

File: publisher/src/components/DataViz/MetricsDataChart.tsx (hunk 2 above)
Comment (nasaownsky) [reply publisher/src/components/DataViz/MetricsDataChart.tsx:254]: Yes, they are!

**Comment by nasaownsky** [comment]
Thank you, @mxosman! Yes, I was able to fix that, check out my changes!

**Comment by mxosman** [comment]
> Thank you, @mxosman! Yes, I was able to fix that, check out my changes!

Ah - thank you so much! Changes look & feel great! 

Sorry, one last thing I didn't notice before until just now -- when you are scrolled down the list and click on the "Switch to Data Table" button, it's showing a bit odd because it's keeping your scroll position. Can we add logic to scroll to top when switching to that view/any other ideas you have?

https://github.com/Recidiviz/justice-counts/assets/59492998/0350a4aa-e878-405c-a546-2978a8dded17

**Comment by nasaownsky** [comment]
@mxosman Oh, great catch, thank you! Fixed :)

**Review by mxosman — APPROVED** [review]
Thank you so much! Looks fantastic to me!

**Pull request merged**